backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py

```diff
@@ -38,3 +30,62 @@ def get_package_info(name: str, version: str) -> dict:
         else:
             LOG.error("Unable to find package info for %s: HTTP %s", name, r.status_code)
             return {}
+
+
+def get_package_license(package_info: dict, package_name: str) -> list[str]:
+    # Official Spec: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#license
+    # We support the official spec, as well as the "deprecated" syntax with objects/arrays in a
+    # `license` or `licenses` property
+    #
+    # So, we should be able to parse all of these:
+    # license: "GPL-3.0"
+    # license: "(GPL-3.0 OR MIT)" # SPDX expression. We will return the expression whole as a string
+    # license: [
+    #     "GPL-3.0",
+    #     "MIT"
+    # ]
+    # license: {
+    #     type: "GPL-3.0",
+    #     url: "..."
+    # }
+    # licenses: [
+    #     { type: "GPL-3.0", url: "..." },
+    #     { type: "MIT", url: "..." }
+    # ]
+
+    if "license" in package_info:
+        license = package_info["license"]
+    elif "licenses" in package_info:
+        license = package_info["licenses"]
+    else:
+        LOG.debug('No license information in package info for "%s"', package_name)
+        return []
+
+    if type(license) is list:
+        result = []
+
+        for item in license:
+            item_license = get_license(item, package_name)
+
+            if item_license:
+                result.append(item_license)
+
+        return result
+    else:
+        item_license = get_license(license, package_name)
+
+        if item_license:
+            return [item_license]
+        else:
+            return []
+
+
+def get_license(item: Union[dict, str], package_name: str) -> Optional[str]:
+    if type(item) is str:
+        return item.lower()
+    elif type(item) is dict:
+        if "type" in item and type(item["type"]) is str:
+            return item["type"].lower()
+
+    LOG.warning('Could not parse a license for "%s". It may be unconventional.', package_name)
```

Comment:
Is warning the right log level for this?

Essentially, I wanted some log to point out if conventions change or if there is a common convention we miss. However, this input is untrusted, so an NPM package messing up the convention or similar will also trigger this path.

Reply:
A warning is perfect here